src/mds/MDSMap.cc

```diff
@@ -604,6 +610,9 @@ void MDSMap::mds_info_t::decode(bufferlist::const_iterator& bl)
   if (struct_v >= 9) {
     decode(flags, bl);
   }
+  if (struct_v >= 10) {
+    decode(compat, bl);
```

Comment:
It doesn't look to me like we're initializing ```compat``` with a good default value in the constructor or here when we decode an older MDSMap?

Reply:
Hmm, you're right. Thanks. I'll fix it.

Reply:
Fixed